stubs/10.0.0/HasManyThrough.stub

```diff
@@ -53,16 +53,25 @@ class HasManyThrough extends Relation
      *
      * @param  mixed  $id
      * @param  array<int, string>  $columns
-     * @return TRelatedModel|\Illuminate\Database\Eloquent\Collection<int, TRelatedModel>|null
+     * @return ($id is (\Illuminate\Contracts\Support\Arrayable<array-key, mixed>|array<mixed>) ? \Illuminate\Database\Eloquent\Collection<int, TRelatedModel> : TRelatedModel|null)
```

Comment:
Oh wow, that is super nice!

Reply:
Yeah, it cleans up a lot of phpstan extensions that were used for this purpose!